pypeit/scripts/show_1dspec.py

```diff
@@ -35,38 +31,28 @@ def main(args):
     """ Runs the XSpecGui on an input file
     """
 
-    try:
-        # TODO: JFH There is a bug here. This should fault if this is not a specobj file, but it does not. So
-        # I've added this ugly hack
-        sobjs = specobjs.SpecObjs.from_fitsfile(args.file)
-        if np.sum(sobjs.OPT_WAVE) is None:
-            raise ValueError("This is an ugly hack until the DataContainer bug is fixed")
-    except:
-        # place holder until coadd data model is sorted out
-        wave, flux, flux_ivar, flux_mask, meta_spec, head = general_spec_reader(args.file)
-        spec = XSpectrum1D.from_tuple((wave*u.AA, flux, np.sqrt(utils.inverse(flux_ivar))), masking='none')
-    else:
-        # List only?
-        if args.list:
-            print("Showing object names for input file...")
-            for ii in range(len(sobjs)):
-                name = sobjs[ii].NAME
-                print("EXT{:07d} = {}".format(ii+1, name))
-            return
+    sobjs = specobjs.SpecObjs.from_fitsfile(args.file)
+    # List only?
+    if args.list:
+        print("Showing object names for input file...")
+        for ii in range(len(sobjs)):
+            name = sobjs[ii].NAME
+            print("EXT{:07d} = {}".format(ii+1, name))
+        return
 
-        if args.obj is not None:
-            exten = np.where(sobjs.NAME == args.obj)[0][0]
-            if exten < 0:
-                msgs.error("Bad input object name: {:s}".format(args.obj))
-        else:
-            exten = args.exten-1 # 1-index in FITS file
+    if args.obj is not None:
+        exten = np.where(sobjs.NAME == args.obj)[0][0]
+        if exten < 0:
+            msgs.error("Bad input object name: {:s}".format(args.obj))
+    else:
+        exten = args.exten-1 # 1-index in FITS file
 
-        # Check Extraction
-        if args.extract == 'OPT':
-            if sobjs[exten]['OPT_WAVE'] is None: #not in sobjs[exten]._data.keys():
-                    msgs.error("Spectrum not extracted with OPT.  Try --extract=BOX")
+    # Check Extraction
```

Comment:
@profxj if you want this ugly hack to work for QL, just change OPT_WAVE to BOX_WAVE for now. Again, this hack is terrible, but not much I can do when the reader is reading in garbage rather than raising an exception as it should.